File: smalivm/src/test/java/org/cf/smalivm/opcode/MonitorEnterOpTest.java
//package org.cf.smalivm.opcode;
//
//import static org.junit.jupiter.api.Assertions.assertEquals;
//
//import org.cf.smalivm.VMTester;
//import org.cf.smalivm.VirtualMachine;
//import org.cf.smalivm.context.ExecutionGraph;
//import org.cf.smalivm.context.ExecutionNode;
//import org.junit.jupiter.api.Test;
//
//public class MonitorEnterOpTest {
//    private static final String CLASS_NAME = "Lmonitor_enter_test;";
//
//    @Test
//    public void monitorEnterHasExpectedTostring() {
//        String methodDescriptor = "monitorEnter()V";
//        VirtualMachine vm = VMTester.spawnVM();
//        ExecutionGraph graph = vm.spawnInstructionGraph(CLASS_NAME, methodDescriptor);
//        ExecutionNode node = graph.getTemplateNode(1);
//        assertEquals("monitor-enter r0", node.getOp().toString());
//    }
//}

File: smalivm/src/test/java/org/cf/smalivm/opcode/MonitorEnterOpTest.kt
package org.cf.smalivm.opcode

import io.mockk.every
import io.mockk.mockk
import org.jf.dexlib2.Opcode
import org.jf.dexlib2.builder.MethodLocation
import org.jf.dexlib2.builder.instruction.BuilderInstruction11x
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

class MonitorEnterOpTest {
    @Test
    fun monitorEnterHasExpectedToString() {
        val instruction: BuilderInstruction11x = mockk {
            every { registerA } returns 3
            every { opcode } returns Opcode.MONITOR_ENTER
        }
        val location: MethodLocation = mockk()
        every { location.instruction } returns instruction
        val op = MonitorEnterOp.build(location, mockk(), mockk(), mockk())
        assertEquals("monitor-enter r3", op.toString())
    }
}
